### PR TITLE
Fix: Running Test with a comment fails test on Understand The immediately invoked function expression iife

### LINKED ISSUE #39346: Running test with a comment at the end makes the test fail.
**Describe your problem and how to reproduce it:**
Running the test will fail if you have anything commented after the code.
Deleting the comment line allows us to pass the test. 
The presence of a comment should not affect the output of the test. 

```
(function() {
  console.log("A cozy nest is ready");
})();
// makeNest();
```
**Add a Link to the page with the problem:**
**Object Oriented Programming: Understand the Immediately Invoked Function Expression (IIFE)**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-immediately-invoked-function-expression-iife

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: Version 84.0.4147.105 (Official Build) (64-bit)
* Operating System: Catalina 10.15.6

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by JoshQuaintance**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39346 

- [x] Regex now checks if the function is an IIFE function
- [x] Passes all 4 solutions

### CODE REVIEW

**Review by naomi-lgbt — DISMISSED** [review]
Tested locally - issue seems to be fixed. Nice work @JoshuaPelealu and congratulations on your first PR! 

Changes LGTM 👍 - as an aside, it is best to [create a branch, work on your changes there, and submit the PR from your branch](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally?id=making-changes-locally) rather than working off the `master` branch of your fork. 🙂

**Comment by JoshuaPelealu** [comment]
Thanks @nhcarrigan... 

I just realized that I made the changes in the master branch. I forgot to branch, but I'll make sure to do it next time so I don't lose any progress.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-immediately-invoked-function-expression-iife.english.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ tests:
   - text: The function should be anonymous.
     testString: assert(/\((function|\(\))(=>|\(\)){?/.test(code.replace(/\s/g, "")));
   - text: Your function should have parentheses at the end of the expression to call it immediately.
-    testString: assert(/}?\)\(\)$|}\(\)\)$/.test(code.replace(/[\s;]/g, "")));
+    testString: assert(/\(.*\)\(\)|\(.*\}\(\)\)/.test(code.replace(/[\s;]/g, "")));
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-immediately-invoked-function-expression-iife.english.md:36]:
```suggestion
    testString: assert(/\(.*(\)\(|\}\(\))\).test(code.replace(/[\s;]/g, "")));
```

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-immediately-invoked-function-expression-iife.english.md:36]: @JoshuaPelealu What do you think about this slightly more DRY version of your regex?

Comment (JoshuaPelealu) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/understand-the-immediately-invoked-function-expression-iife.english.md]: I like it. It's more structured.

**Review by naomi-lgbt — APPROVED** [review]
LGTM!

**Review by moT01 — APPROVED** [review]
LGTM @JoshuaPelealu 🎉 Congratulations on your first contribution to freeCodeCamp 🎉

**Pull request merged**